crates/common_types/src/payments.rs

```diff
@@ -38,3 +40,35 @@ pub struct StripeSplitPaymentRequest {
     pub transfer_account_id: String,
 }
 impl_to_sql_from_sql_json!(StripeSplitPaymentRequest);
+
+#[derive(
+    Serialize, Deserialize, Debug, Clone, PartialEq, Eq, FromSqlRow, AsExpression, ToSchema,
+)]
+#[diesel(sql_type = Jsonb)]
+#[serde(deny_unknown_fields)]
+/// Hashmap to store mca_id's with product names
+pub struct AuthenticationConnectorAccountMap(
+    HashMap<enums::AuthenticationProduct, common_utils::id_type::MerchantConnectorAccountId>,
+);
+impl_to_sql_from_sql_json!(AuthenticationConnectorAccountMap);
+
+impl AuthenticationConnectorAccountMap {
+    /// get inner hashmap
+    pub fn inner(
```

Comment:
Ideally, we would never need to access inner value directly. Can remove this if not required